google/cloud/spanner_v1/snapshot.py

```diff
@@ -101,7 +108,13 @@ def _restart_on_unavailable(
                     observability_options=observability_options,
                     metadata=metadata,
                 ), MetricsCapture():
-                    iterator = method(request=request, metadata=metadata)
+                    attempt += 1
```

Comment:
This isn't really needed, as this call will only be used the first time (due to the `if iterator is None` check above)